darwin/torch/dataset.py

```diff
@@ -14,72 +16,17 @@
 )
 
 
-def get_dataset(
```

Comment:
After giving it some thoughts, I don't think getting rid of this function is a good idea.

My major arguments are:
1. The logic to fetch the dataset from darwin has been moved to the dataset classes' constructors. I think that our Datasets here should just wrap around Pytorch ones to load images, and load the annotations according to the _mode_ we're at. Basically, they exist on a lower level than the user's API to fetch the data.
2. To add a further point on top of that: supposing that this code is going to be used by Python coders that are not 100% knowledgeable about the internals of Pytorch. Then, having a unique function to know about is much easier than knowing that, since you're doing instance segmentation, you need a specific dataset.

Please let me know if you think these points make sense, as this discussion is rather important to have sooner rather than later, so we can be all on the same page.

@simedw @seenitall

Reply:
I think the logic of downloading the dataset should not be in the dataset itself. The dataset itself should be a Pytorch-like dataset which only provides the logic of loading the data and preparing it for the dataloader to fetch it trough `__get_item__()`. 
The `get_dataset` function behaves very much like a factory method pattern and I think its a clean and scalable solution should there be another type of datasets in the future.